app/services/roster_entry_sorter.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Sorts roster entries by:
+# - Linked + accepted
+# - Linked + not accepted
+# - Not linked
+
+class RosterEntrySorter
+  attr_reader :entries, :assignment
+
+  def initialize(roster_entries, assignment)
+    @entries = roster_entries
+    @assignment = assignment
+  end
+
+  def sort
+    users_with_repo = @assignment.repos.map(&:user_id)
```

Comment:
Can use use [`#pluck`](http://guides.rubyonrails.org/active_record_querying.html#pluck) here? That way we don't have to load all of the data for each record?

```ruby
users_with_repo = @assignment.repos.pluck(:user_id)
```